mesa/visualization/frontend/src/initGUI.js

```diff
@@ -5,185 +5,192 @@ export const initGUI = function (model_params, send) {
-            ticks_positions: [0, 100],
-        });
-        sliderInput.on("change", function () {
-            onSubmitCallback(param, Number($(this).val()));
-        });
+      const domID = param + "_id";
+      const tooltipID = domID + "_tooltip";
+      let tooltip = "";
+      // Enable tooltip label
+      if (obj.description !== null) {
+        tooltip = `title='${obj.description}'`;
+      }
+  
+      sidebar.append(
+        [
+          "<div>",
+          "<p>",
+          `<span id='${tooltipID}' ${tooltip} data-bs-toggle='tooltip' data-bs-placement='top' class='badge bg-primary'>`,
+          obj.name,
+          "</span>",
+          "</p>",
+          "<input id='" + domID + "' type='text' />",
+          "</div>",
+        ].join("")
+      );
+  
+      // Setup slider
+      const sliderInput = new Slider("#" + domID, {
+        min: obj.min_value,
+        max: obj.max_value,
+        value: obj.value,
+        step: obj.step,
+        ticks: [obj.min_value, obj.max_value],
+        ticks_labels: [obj.min_value, obj.max_value],
+        ticks_positions: [0, 100],
+      });
+      document.getElementById(domID).onchange = () => {
+        onSubmitCallback(param, Number(sliderInput.getValue()));
+      };
     };
-
+  
     const addChoiceInput = function (param, obj) {
-        const domID = param + "_id";
-        const span = "<span class='caret'></span>";
-        const template = [
-            "<p><label for='" +
-            domID +
-            "' class='badge badge-primary'>" +
-            obj.name +
-            "</label></p>",
-            "<div class='dropdown'>",
-            "<button id='" +
-            domID +
-            "' class='btn btn-default dropdown-toggle' type='button' data-toggle='dropdown'>" +
-            obj.value +
-            " " +
-            span,
-            "</button>",
-            "<ul class='dropdown-menu' role='menu' aria-labelledby='" + domID + "'>",
-        ];
-        const choiceIdentifiers = [];
-        for (let i = 0; i < obj.choices.length; i++) {
-            const choiceID = domID + "_choice_" + i;
-            choiceIdentifiers.push(choiceID);
-            template.push(
-                "<li role='presentation'><a class='pick-choice' id='" +
-                choiceID +
-                "' role='menuitem' tabindex='-1' href='#'>",
-                obj.choices[i],
-                "</a></li>"
-            );
-        }
-
-        // Close the dropdown options
-        template.push("</ul>", "</div>");
-
-        // Finally render the dropdown and activate choice listeners
-        sidebar.append(template.join(""));
-        choiceIdentifiers.forEach(function (id, idx) {
-            $("#" + id).on("click", function () {
-                const value = obj.choices[idx];
-                $("#" + domID).html(value + " " + span);
-                onSubmitCallback(param, value);
-            });
+      const domID = param + "_id";
```

Comment:
The `Choice` implementation has been made much simpler in https://github.com/projectmesa/mesa/commit/7b3b072ed1167666110339f1ec39daeb8a67547e.

Reply:
Huh, thanks. I was actually waiting for this commit to be merged, but I must have made a mistake while updating my files with your changes.
Anyhow I will evolve this pr more to be more idiomatic react code and not the current mixture of declarative and Imperative code. So this is all temporary

Reply:
It's more of with the change from HTML dropdown to HTML select. The latter is much simpler. Dropdown should be used only when each dropdown items has different event listener.